lib/broadway.ex

```diff
@@ -905,7 +905,32 @@ defmodule Broadway do
   @doc since: "0.5.0"
   @callback handle_failed(messages :: [Message.t()], context :: term) :: [Message.t()]
 
-  @optional_callbacks prepare_messages: 2, handle_batch: 4, handle_failed: 2
+  @doc """
+  Invoked to get the process name of this Broadway pipeline.
+
+  `broadway_name` is the name given to `start_link/2` in the `:name` option. `base_name`
+  is a string used by Broadway to identify different components of the pipeline
+  whose name needs to be registered (such as "batcher" or "processor").
+
+  The return value of this callback must be a process name that is valid for registration.
+  See the name registration rules in the documentation for `GenServer`.
+
+  This callback is optional. If not defined, the `broadway_name` given to `start_link/2`
+  **must be an atom**: the default implementation of this callback will fail otherwise.
+
+  ## Examples
+
+      @impl Broadway
+      def process_name({:via, module, term}, base_name) do
+        {:via, module, {term, base_name}}
+      end
+
+  """
+  @doc since: "1.1.0"
```

Comment:
Had to put 1.1.0 here because it's never been available publicly before and I needed a minor version bump to introduce a new feature.